src/compojure/api/meta.clj

```diff
@@ -531,14 +532,14 @@
 
 (defmacro static-context
   [path route]
-  `(#'compojure.core/if-context
+  `(#'compojure.api.compojure-compat/make-context
      ~(#'compojure.core/context-route path)
-     ~route))
+     (constantly ~route)))
 
 (defn routing [handlers]
   (if-let [handlers (seq (keep identity handlers))]
-    (apply some-fn handlers)
-    (constantly nil)))
+    (apply compojure.core/routes handlers)
```

Comment:
I don't see a way to inline it that would work with both sync and async handlers, so I decided to keep it simple.